client/verta/verta/_internal_utils/http_session_utils.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+""" Utilities for working with request.Session objects, and handling of retries """
+
+# TODO: Unit test for prediction_input_unpack method
+# TODO: Unit test for prediction_io_cleanup method
+import requests
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+from typing import Dict, Optional, Set
+from urllib3.util import Retry
+
+DEFAULT_MAX_RETRIES: int = 13
+DEFAULT_BACKOFF_FACTOR: float = 0.3
+DEFAULT_STATUS_FORCELIST: Set = {404, 429}
+
+
+def retry_config(max_retries: int,
+                 status_forcelist: Set[int],
+                 backoff_factor: float) -> Retry:
+    """ Return a Retry object with the given parameters """
+    return Retry(total=None, # Rely on `status` param to set the number of retries on failed status codes
+                 connect=0,  # Connection errors before the request is sent to the remote server should raise an error
+                 status=max_retries,
+                 other=0,  # recommended by urllib3 docs to guard against infinite retry loops
+                 status_forcelist=status_forcelist,
+                 backoff_factor=backoff_factor,
+                 allowed_methods=False
+                 )
+
+
+def alter_retry_config(session: Session,
+                       max_retries: Optional[int],
+                       status_forcelist: Optional[Set[int]],
+                       backoff_factor: Optional[float]) -> Session:
```

Comment:
nit: should we provide default `None` for these?

Reply:
Upon closer look, I think these shouldn't actually be `Optional`.  The predict methods are always going to use defaults for those values when calling `update_retry_config`.

Reply:
Pending commit will drop `Optional` from type declaration.